tests/MongoDB.Driver.Tests/UnifiedTestOperations/UnifiedAssertSessionUnpinnedOperation.cs

```diff
@@ -0,0 +1,66 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Driver.Tests.Specifications.unified_test_format;
+
+namespace MongoDB.Driver.Tests.UnifiedTestOperations
+{
+    public class UnifiedAssertSessionUnpinnedOperation : IUnifiedSpecialTestOperation
+    {
+        private readonly IClientSessionHandle _session;
+
+        public UnifiedAssertSessionUnpinnedOperation(IClientSessionHandle session)
+        {
+            _session = session;
+        }
+
+        public void Execute()
+        {
+            _session.WrappedCoreSession?.CurrentTransaction?.PinnedServer.Should().BeNull();
```

Comment:
Consider:

            _session.WrappedCoreSession.CurrentTransaction.PinnedServer.Should().BeNull();

Do you want the test to pass if CurrentTransaction is null?

I would expect that any assertion against the session being unpinned would assume that we ARE in a transaction, but maybe that's not true...